tempto-core/src/main/java/com/teradata/tempto/internal/convention/ConventionBasedTestProxyGenerator.java

```diff
@@ -81,7 +87,23 @@ public ConventionBasedTest generateProxy(ConventionBasedTest conventionBasedTest
 
     private String generatedClassName(ConventionBasedTest conventionBasedTest)
     {
-        return testPackage + "." + capitalize(replace(conventionBasedTest.testName(), "-", "_"));
+        List<String> testNameParts = Splitter.on('.').splitToList(conventionBasedTest.getTestName());
+        return testPackage + "." + toJavaSymbol(testNameParts.get(testNameParts.size() - 2));
+    }
+
+    private String generatedMethodName(ConventionBasedTest conventionBasedTest)
+    {
+        List<String> testNameParts = Splitter.on('.').splitToList(conventionBasedTest.getTestName());
+        return toJavaSymbol(Iterables.getLast(testNameParts));
+    }
+
+    private String toJavaSymbol(String s)
+    {
+        String javaSymbol = s.replaceAll("[^A-Z-a-z0-9_]", "");
```

Comment:
do I understand correctly that '-' is accepted as a java symbol character? In such case I think it is incorrect.